packages/flutter/test/widgets/semantics_tester_generateTestSemanticsExpressionForCurrentSemanticsTree_test.dart

```diff
@@ -0,0 +1,139 @@
+      .split('\n')
+      .map((String line) => line.trim())
+      .join('\n')
+      .trim() + ',';
+
+    File findThisTestFile(Directory directory) {
+      for (FileSystemEntity entity in directory.listSync()) {
+        if (entity is Directory) {
+          final File childSearch = findThisTestFile(entity);
+          if (childSearch != null) {
+            return childSearch;
+          }
+        } else if (entity is File && entity.path.endsWith('semantics_tester_generateTestSemanticsExpressionForCurrentSemanticsTree_test.dart')) {
+          return entity;
+        }
+      }
+      return null;
+    }
+
+    final File thisTestFile = findThisTestFile(Directory.current);
+    expect(thisTestFile, isNotNull);
+    String expectedCode = thisTestFile.readAsStringSync();
+    expectedCode = expectedCode.substring(
+      expectedCode.indexOf('>' * 12) + 12,
+      expectedCode.indexOf('<' * 12) - 3,
+    )
+      .split('\n')
+      .map((String line) => line.trim())
+      .join('\n')
+      .trim();
+    semantics.dispose();
+    expect(code, expectedCode);
+  });
+
+  testWidgets('generated code is correct', (WidgetTester tester) async {
+    final SemanticsTester semantics = new SemanticsTester(tester);
+    await pumpTestWidget(tester);
+    expect(
+      semantics,
+      hasSemantics(
+        // The code below delimited by > and < characters is generated by
+        // generateTestSemanticsExpressionForCurrentSemanticsTree function.
+        // You must update it when changing the output generated by
+        // generateTestSemanticsExpressionForCurrentSemanticsTree. Otherwise,
+        // the test 'generates code', defined above, will fail.
+        // >>>>>>>>>>>>
+        new TestSemantics(
+          children: <TestSemantics>[
+            new TestSemantics(
+              children: <TestSemantics>[
+                new TestSemantics(
+                  children: <TestSemantics>[
+                    new TestSemantics(
+                      label: r'Plain text',
+                      textDirection: TextDirection.ltr,
+                    ),
+                    new TestSemantics(
+                      flags: <SemanticsFlags>[SemanticsFlags.hasCheckedState, SemanticsFlags.isChecked, SemanticsFlags.isSelected],
+                      actions: <SemanticsAction>[SemanticsAction.tap, SemanticsAction.decrease],
+                      label: r'‪Interactive text‬',
+                      value: r'test-value',
+                      increasedValue: r'test-increasedValue',
+                      decreasedValue: r'test-decreasedValue',
+                      hint: r'test-hint',
+                      textDirection: TextDirection.rtl,
+                    ),
+                  ],
+                ),
+              ],
+            ),
+          ],
+        ),
+        // <<<<<<<<<<<<
+        ignoreRect: true,
+        ignoreTransform: true,
+        ignoreId: true,
+      )
+    );
+    semantics.dispose();
+  });
+}
```

Comment:
First, this is the kind of crazy gymnastics code that I love to write. My hat is off to you sir. Never show this code to @amirh.

Second, can I ask you to use something other that "<<<" / ">>>" to delimit the code snippet? <<</>>> clashes with merge markers and that might cause hassle later.